src/main/scala/scala/tools/refactoring/common/TreeTraverser.scala

```diff
@@ -119,7 +119,15 @@ trait TreeTraverser extends TracingImpl {
       }
 
       val treesFromType = {
-        val tpeWithoutPrefix = tpe.trimPrefix(tpe.toString).split("\\.") match {
+        def splitUntilTypeParameterBracket(acc: Array[String])(tpe: String): Array[String] = {
+          if (tpe.isEmpty)
+            acc
+          else if (!tpe.contains(".") || tpe.contains(".") && tpe.contains("[") && tpe.indexOf("[") < tpe.indexOf("."))
+            acc :+ tpe
```

Comment:
I'm sorry, but you provoked me with your comment about Sauron :-) Try organizing imports on
```scala
object `Eye[` {
  object `.Of` {
    object Sauron
    object TheBeholder
  }
}

class Bug {
  import `Eye[`.`.Of`.TheBeholder
  import `Eye[`.`.Of`.Sauron

  val x = Sauron
}
```